crates/nu-path/src/helpers.rs

```diff
@@ -6,27 +8,29 @@ pub fn home_dir() -> Option<AbsolutePathBuf> {
 
 /// Return the data directory for the current platform or XDG_DATA_HOME if specified.
 pub fn data_dir() -> Option<AbsolutePathBuf> {
-    std::env::var("XDG_DATA_HOME")
-        .ok()
-        .and_then(|path| AbsolutePathBuf::try_from(path).ok())
-        .or_else(|| dirs::data_dir().and_then(|path| AbsolutePathBuf::try_from(path).ok()))
-        .map(|path| path.canonicalize().map(Into::into).unwrap_or(path))
+    configutable_dir_path("XDG_DATA_HOME", dirs::data_dir)
```

Comment:
All these are spelled wrong.
```suggestion
    configurable_dir_path("XDG_DATA_HOME", dirs::data_dir)
```